### PR TITLE
Add sig-network 2021 report

### PR DESCRIPTION
**Pull request by caseydavenport**
Add the SIG network annual report for 2021.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/community/issues/6375

### CODE REVIEW

File: sig-network/annual-report-2021.md [hunk 1]
```diff
@@ -4,140 +4,112 @@
...
+- Unique reviewers for SIG-owned packages: 29
+- Unique approvers for SIG-owned packages: 29
 
 ## Subprojects
 
-<!--
-In future, this will be generated from delta of sigs.yaml from $YYYY-01-01 to $YYYY-12-31
-Manually visible via `git diff HEAD@{$YYYY-01-01} HEAD@{$YYYY-12-31} -- $sig-id/README.md`
--->
+New in $2021:
```
Comment (aojea) [inline sig-network/annual-report-2021.md:80]: ?
```suggestion
New in 2021:
```

File: sig-network/annual-report-2021.md [hunk 2]
```diff
@@ -4,140 +4,112 @@
 
 1. What work did the SIG do this year that should be highlighted?
 
-   -
-   -
-   -
+- [Dual-stack IPv4/IPv6 networking feature reaches GA](https://kubernetes.io/blog/2021/12/08/dual-stack-networking-ga/) 
+- [KPNG](https://github.com/kubernetes-sigs/kpng)
+- [AdminNetworkPolicy KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2091-admin-network-policy)
+- [Gateway API](https://gateway-api.sigs.k8s.io/)
```
Comment (markmc) [inline sig-network/annual-report-2021.md:10]:
```suggestion
- Service APIs was renamed to [Gateway API](https://gateway-api.sigs.k8s.io/), has progressed to [v1alpha2](https://gateway-api.sigs.k8s.io/blog/2021/introducing-v1alpha2/), and is now supported by a dozen [implementations](https://gateway-api.sigs.k8s.io/implementations).
```

FYI @robscott

File: sig-network/annual-report-2021.md [hunk 3]
```diff
@@ -4,140 +4,112 @@
...
 
 2. What initiatives are you working on that aren't being tracked in KEPs?
 
-   -
-   -
-   -
+- Gateway API: https://gateway-api.sigs.k8s.io/
```
Comment (markmc) [inline sig-network/annual-report-2021.md:14]:
```suggestion
- [Gateway API](https://gateway-api.sigs.k8s.io/) has defined its own [GEP process](https://gateway-api.sigs.k8s.io/contributing/gep/) for tracking enhancements. See below for some 2021 highlights. The group is currently working towards its upcoming [v1beta1 milestone](https://github.com/kubernetes-sigs/gateway-api/issues?q=is%3Aopen+is%3Aissue+milestone%3Av1beta1).
```

@robscott

File: sig-network/annual-report-2021.md [hunk 4]
```diff
@@ -4,140 +4,112 @@
...
+     - [1669 - Proxy terminating endpoints](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1669-proxy-terminating-endpoints) 
+
    - Alpha
-     - [$kep-number - $title](https://git.k8s.io/community/$link/README.md) - $milestone.alpha
-     - [$kep-number - $title](https://git.k8s.io/community/$link/README.md) - $milestone.alpha
-   - Pre-alpha
-     - [$kep-number - $title](https://git.k8s.io/community/$link/README.md)
+
+     - [1435 - Mixed protocol LB](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1435-mixed-protocol-lb)
+     - [2595 - Expanded DNS config](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2595-expanded-dns-config)
```
Comment (markmc) [inline sig-network/annual-report-2021.md:36]:
```suggestion
Gateway API Enhancements:

* v1alpha2 highlights:
    * Transitioned to new `gateway.networking.k8s.io` API group to reflect its status as an official Kubernetes API.
    * [GEP-724](https://gateway-api.sigs.k8s.io/geps/gep-724/): Simpler Route-Gateway binding.
    * [GEP-709](https://gateway-api.sigs.k8s.io/geps/gep-709/): Safe cross namespace references.
    * [GEP-713](https://gateway-api.sigs.k8s.io/geps/gep-713/): Policy attachment.
* Other notable initiatives:
    * [GEP-917](https://gateway-api.sigs.k8s.io/geps/gep-917/): Gateway API conformance testing.
    * [GEP-922](https://gateway-api.sigs.k8s.io/geps/gep-922/): Gateway API conformance versioning.
    * [Validating admission webhook](https://github.com/kubernetes-sigs/gateway-api/issues/487).
```

@robscott

**Review by markmc — COMMENTED** [review]
Just some suggestions for Gateway API content. Best if @robscott review these before accepting

**Comment by aojea** [comment]
/lgtm

we are half way into 2022 XD

**Pull request merged**